consul-lambda-registrator/upsert_event.go

```diff
@@ -67,25 +74,74 @@ func (env Environment) storeServiceDefaults(e UpsertEvent) error {
 
 	// There is no need for CAS because we are completely regenerating the service
 	// defaults config entry.
-	_, _, err := env.ConsulClient.ConfigEntries().Set(serviceDefaults, e.writeOptions())
+	_, _, err := env.ConsulClient.ConfigEntries().Set(serviceDefaults, e.WriteOptions())
 
 	return err
 }
 
-func (e UpsertEvent) writeOptions() *api.WriteOptions {
-	var writeOptions *api.WriteOptions
-	if e.EnterpriseMeta != nil {
-		writeOptions = &api.WriteOptions{
-			Partition: e.EnterpriseMeta.Partition,
-			Namespace: e.EnterpriseMeta.Namespace,
+func (env Environment) upsertTLSData(e UpsertEvent) error {
+	if !env.IsManagingTLS() {
+		return nil
+	}
+
+	env.Logger.Debug("upserting mTLS data", "service", e.Name)
+
+	// Retrieve Consul root CA
+	// TODO: Should optimize this flow so that we only request the root CA once, not for every service.
```

Comment:
I think it's probably worth optimizing this soon. It will save a bunch of API calls.

Reply:
OK, I will do this in a follow up PR.